README.md

```diff
@@ -1,35 +1,133 @@
+
+```console
+git push heroku HEAD:master
+```
+
+You will see the buildlog for your application, followed by a message about successful
+deployment.
+
+When you now navigate to your application on Heroku (following the name from the example it would
+be at `https://ancient-woodland-33672.herokuapp.com/`) you will be prompted to login with Github.
+
+After successful authentication, Github will redirect you back to your application and a success
+page will be shown.
+
+For futher steps, you might want to have a look in the [configuration Section](#configuration) to
+learn about configuration options for finer grained authentication, e.g. allowing only members
+of a particular Github organization.
+
+## Usage with other applications
+
+For Using this buildpack with your application, you need to do two things:
+
+On the one hand, you need to set up the configuration for `oauth2_proxy`. The getting started section
+describes this process for Github. For other providers, a look at the
+[configuration Section](#configuration) and at the
+[OAuth Provider Configuration documentation of oauth2\_proxy](https://github.com/pusher/oauth2_proxy#oauth-provider-configuration)
+will provide you with the necessary information.
+
+On the other hand, you need to ensure that `oauth2_proxy` is run as a reverse proxy in front
+of your actual application worker.
+
+For that purpose, this buildpack installs a script `start_with_oauth2_proxy.sh`. This script
+can be used as the `web` process and will pass any requests to the backend process specified as
+its arguments.
+
+For example, the `heroku-buildpack-static` has `/app/bin/boot` as entrypoint. Given that,
+a usable `Procfile` to run it with the proxy looks as following:
 
 ```console
 web: /app/bin/start_with_oauth2_proxy.sh /app/bin/boot
 ```
 
+This will take care to start both `/app/bin/boot` and `oauth2_proxy` and to route any incoming
+requests correctly.
+
 ## Configuration
 
 The following environment variables are required:
 
 - `OAUTH2_PROXY_PROVIDER`: The provider to use. Something like `github`, `google` or `facebook`
-- `OAUTH2_PROXY_CLIENT_ID`: The oauth2 client id (generated by auth provider)
-- `OAUTH2_PROXY_CLIENT_SECRET`: The oauth2 client secret (generated by auth provider)
-- `OAUTH2_PROXY_COOKIE_SECRET`: Secret key to encrypt `oauth2_proxy`'s session cookies
+- `OAUTH2_PROXY_CLIENT_ID`: The OAuth2 client id (generated by auth provider)
+- `OAUTH2_PROXY_CLIENT_SECRET`: The OAuth2 client secret (generated by auth provider)
+- `OAUTH2_PROXY_COOKIE_SECRET`: Secret key to encrypt `oauth2_proxy`'s session cookies. This string
+  needs to be 32 characters long.
 
 Optionally, you can provide the following:
 
-- `OAUTH2_GITHUB_ORG`: Only allows login for members of the given github organization
-- `OAUTH2_EMAIL_DOMAIN`: Only allows login for users with emails from this domain
+- `OAUTH2_GITHUB_ORG`: Only allow login for members of the given github organization
```

Comment:
Here you need to decide to use "Github" like in the rest of the document or "GitHub" if you decide to update the other appearances.